tests/frontend/test_prepost.py

```
@@ -2076,3 +2076,105 @@ def test_streaming_parallel_tool_calls_no_think(
+
+# ---------------------------------------------------------------------------
+# Reasoning-parser adjust_request wiring
+# ---------------------------------------------------------------------------
+class _RecordingReasoningParser:
+    """Stand-in for a reasoning parser whose adjust_request mutates the request.
+
+    Real example: a parser that splits thinking from the answer on text markers needs
+    skip_special_tokens=False so those markers survive detokenisation.
+    """
+
+    instances: list["_RecordingReasoningParser"] = []
```

Ruff (0.16.1)

[warning] Mutable default value for class attribute

(RUF012)